### ISSUE TITLE
Rethink current naming convention in contracts

### ISSUE THREAD

**Issue by bidzyyys** [OpenZeppelin/rust-contracts-stylus#386]
### What is the feature you would like to see?

Right now we use in some cases Solidity like naming convention (`_do_sth()`). It is not recommended attitude in Rust and causes `cargo clippy` issues (right now disabled). We should rethink this topic and make sure our library is consistent in terms of naming.

Related https://github.com/OpenZeppelin/rust-contracts-stylus/issues/491
Related https://github.com/OpenZeppelin/rust-contracts-stylus/issues/499

TODO: Remove `_` prefix from state variables (for functions see https://github.com/OpenZeppelin/rust-contracts-stylus/issues/386#issuecomment-2595763777)

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

**Comment by bidzyyys** [comment]
The same applies for struct attributes, like `_pub _owner: StorageAddress,`

**Comment by 0xNeshi** [comment]
Do we include [test name standardization](https://openzeppelin.slack.com/archives/C06R5AGFE9G/p1732709957377129?thread_ts=1732618832.030739&cid=C06R5AGFE9G) in this issue as well?

- `function_name_description` for standard tests.
- `function_name_revert_when_condition` for tests expecting a revert

**Comment by bidzyyys** [comment]
> Do we include [test name standardization](https://openzeppelin.slack.com/archives/C06R5AGFE9G/p1732709957377129?thread_ts=1732618832.030739&cid=C06R5AGFE9G) in this issue as well?
> 
> * `function_name_description` for standard tests.
> * `function_name_revert_when_condition` for tests expecting a revert

@0xNeshi not in this issue, for test name standardisation I suggest having another one. Easier to solve & merge.

**Comment by 0xNeshi** [comment]
Makes sense, created a separate issue https://github.com/OpenZeppelin/rust-contracts-stylus/issues/491

**Comment by 0xNeshi** [comment]
@bidzyyys @qalisander 

In Solidity, internal functions use the `_fn_name` prefix pattern, making their visibility immediately clear. In our Rust contracts, we have a few scenarios to consider:

1. When a public function has an internal implementation (like `transfer_ownership`), we can use the `fn_name_internal` suffix pattern. This maintains clarity while following Rust conventions.

2. However, we also have internal functions meant for inheritance that don't have public counterparts (like `_mint` in ERC20). Simply removing the underscore could be misleading - developers might assume these functions are public since they don't follow the `_internal` suffix pattern they see elsewhere in the codebase.

Here's my proposal: Let's consistently use the `_internal` suffix for all internal functions. This approach would:
- Maintain clear visibility indicators like Solidity
- Follow Rust naming conventions
- Provide consistent patterns across our codebase
- Help developers quickly understand function visibility

What do you think about adopting this consistent pattern? It seems like a good balance between maintaining familiarity for Solidity developers while embracing Rust's conventions.

**Comment by 0xNeshi** [comment]
We agreed to keep `_` prefix for internal functions, until the moment stylus sdk with advanced inheritance is released.
Then we will reconsider.
